.azure-pipelines.yml

```diff
@@ -26,6 +25,77 @@ jobs:
+      # the first parent is the target branch (usually master) and the second
+      # parent the PR branch.
+      echo Build.SourceBranch=$(Build.SourceBranch)
+      echo Build.SourceBranchName=$(Build.SourceBranchName)
+      echo Build.SourceVersion=$(Build.SourceVersion)
+      echo System.PullRequest.SourceBranch=$(System.PullRequest.SourceBranch)
+      echo System.PullRequest.TargetBranch=$(System.PullRequest.TargetBranch)
+      MERGE_COMMIT=`git rev-parse HEAD`
+      echo MERGE_COMMIT=$MERGE_COMMIT
+      FIRST_PARENT=`git rev-parse HEAD^1`
+      echo FIRST_PARENT=$FIRST_PARENT
+      SECOND_PARENT=`git rev-parse HEAD^2`
+      echo SECOND_PARENT=$SECOND_PARENT
+      echo "##vso[task.setvariable variable=Merge;isOutput=true]$MERGE_COMMIT";
+      echo "##vso[task.setvariable variable=TargetBranch;isOutput=true]$FIRST_PARENT";
+      echo "##vso[task.setvariable variable=SourceBranch;isOutput=true]$SECOND_PARENT";
+    name: commits
+    displayName: 'Record commits'
+
+- job: affected_macOS
+  displayName: 'affected tests (Safari Technology Preview)'
+  #dependsOn: root
+  #condition: dependencies.root.outputs['test_jobs.stability']
+  pool:
+    vmImage: 'macOS-10.13'
+  steps:
+  - template: tools/ci/azure/checkout.yml
+  - template: tools/ci/azure/pip_install.yml
+    parameters:
+      packages: virtualenv
+  - template: tools/ci/azure/install_fonts.yml
+  - template: tools/ci/azure/install_certs.yml
+  - template: tools/ci/azure/install_safari.yml
+  - template: tools/ci/azure/update_hosts.yml
+  - script: no_proxy='*' ./wpt run --yes --no-pause --no-fail-on-unexpected --no-restart-on-unexpected --affected HEAD^1 --log-wptreport $(Build.ArtifactStagingDirectory)/wpt_report.json --channel preview safari
+    displayName: 'Run tests'
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish results'
+    inputs:
+      pathtoPublish: '$(Build.ArtifactStagingDirectory)/wpt_report.json'
+      artifactName: 'results'
+    condition: succeededOrFailed()
+
+- job: affected_sans_changes_macOS
+  displayName: 'affected tests without changes (Safari Technology Preview)'
+  #dependsOn: root
+  #condition: dependencies.root.outputs['test_jobs.stability']
+  pool:
+    vmImage: 'macOS-10.13'
+  steps:
+  - template: tools/ci/azure/checkout.yml
+  # check out first parent, which is the target branch
+  - script: git checkout HEAD^1
+    displayName: 'Checkout target branch'
+  - template: tools/ci/azure/pip_install.yml
+    parameters:
+      packages: virtualenv
+  - template: tools/ci/azure/install_fonts.yml
+  - template: tools/ci/azure/install_certs.yml
+  - template: tools/ci/azure/install_safari.yml
+  - template: tools/ci/azure/update_hosts.yml
+  - script: no_proxy='*' ./wpt run --yes --no-pause --no-fail-on-unexpected --no-restart-on-unexpected --affected 'HEAD@{1}' --log-wptreport $(Build.ArtifactStagingDirectory)/wpt_report.json --channel preview safari
```

Comment:
I can run this command just fine locally (of course, not using Safari), but on Azure it's complaining about invalid arguments:

```
$ no_proxy='*' ./wpt run --yes --no-pause --no-fail-on-unexpected --no-restart-on-unexpected --affected 'HEAD@{1}' --log-wptreport /Users/vsts/agent/2.142.1/work/1/a/wpt_report.json --channel preview safari
wpt run: error: unrecognized arguments: --affected safari
```

I don't really understand why this fails (while the "with change" counterpart succeeds).

Reply:
That's because the argument didn't exist before this PR :) Will work when the argument is landed on master first.